### PR TITLE
DietPi-Globals | G_WHIP speed-up, hopefully 😄

### PR DESCRIPTION
**Pull request by MichaIng**
**Status**: Ready

**Reference**: https://github.com/Fourdee/DietPi/issues/2079

**Commit list/description**:
+ DietPi-Globals | G_WHIP speed-up, hopefully 😄

### CODE REVIEW

File: dietpi/func/dietpi-globals [hunk 1]
```diff
@@ -646,79 +642,48 @@ $(ps f -eo pid,user,tty,cmd | grep -i [d]ietpi)"; then
...
 			# - Requires additional line on Y
 			((lines_required_whip_y++))
 
 			#Auto length for ─
 			#	Get max length of all the lines in odd number array 1st | '' 'this one'
 			local i=0
-			local index=0
 			local character_count_max=0
-			for (( i=0; i<${#G_WHIP_MENU_ARRAY[@]}; i++ ))
+			for (( i=0; i<${#G_WHIP_MENU_ARRAY[@]}; i=i+2 ))
```
Comment (Fourdee) [inline dietpi/func/dietpi-globals]: @MichaIng 

Nice 👍 

---

To keep our standards?
```
i=$i+2
```

Another method aswell:
```
for (( i=0; i<10; i+=2 ))
do

echo $i

done
```

```
root@DietPi-ASUSTB:~# ./test
0
2
4
6
8
```

File: dietpi/func/dietpi-globals [hunk 2]
```diff
@@ -646,79 +642,48 @@ $(ps f -eo pid,user,tty,cmd | grep -i [d]ietpi)"; then
...
 			# - Requires additional line on Y
 			((lines_required_whip_y++))
 
 			#Auto length for ─
 			#	Get max length of all the lines in odd number array 1st | '' 'this one'
 			local i=0
-			local index=0
 			local character_count_max=0
-			for (( i=0; i<${#G_WHIP_MENU_ARRAY[@]}; i++ ))
+			for (( i=0; i<${#G_WHIP_MENU_ARRAY[@]}; i+=2 ))
```
Comment (Fourdee) [inline dietpi/func/dietpi-globals:673]: @MichaIng 

Nice 👍

**Comment by Fourdee** [comment]
@MichaIng 

Very nice, great work 👍 

Will merge for testing.

**Comment by Fourdee** [comment]
@MichaIng 

🈴 Missing text:
image

```
			G_WHIP_CHECKLIST "Please use the spacebar to select the software you wish to install.\n - Software and usage details: https://dietpi.com/software\n - NB: Pressing 'ESC' or selecting 'Back' will clear all changed selections"
```

---

Pre-merge:
image

File: dietpi/func/dietpi-globals (hunk 2 above)
Comment (MichaIng) [reply dietpi/func/dietpi-globals:673]: @Fourdee 
Your comment as well. Lol found the `var+=$value` syntax already but did not use it there. Makes arithmetic assignments again simpler 😃.

**Comment by MichaIng** [comment]
@Fourdee 
Whoopsie `for (( i=0; i<${#G_WHIP_MENU_ARRAY[@]}; i+=2 ))` need to be for `(( i=1; i<${#G_WHIP_MENU_ARRAY[@]}; i+=2 ))`, starting with 1 to loop through the correct array entries.

Will send commit: https://github.com/Fourdee/DietPi/commit/6570a9949f84f18d68f8476061f1796814cfb25d

**Pull request merged**